adapters/webui/internal/frontend/home.html

```diff
@@ -0,0 +1,412 @@
+                <option value="desc">DESC</option>
+                <option value="asc">ASC</option>
+            </select>
+        </div>
+
+        <!-- Search Button -->
+        <div class="flex flex-col justify-end">
+            <button onclick="collectAndUpdateTable()" class="bg-blue-600 hover:bg-blue-700 text-white font-semibold py-2 px-4 rounded-lg flex items-center justify-center focus:outline-none focus:ring-2 focus:ring-blue-500 transition duration-150 ease-in-out">
+                <div>Search</div>
+            </button>
+        </div>
+    </div>
+
+    <div class="p-6 bg-white dark:bg-gray-800 rounded-lg shadow-md mt-6">
+        <div class="flex justify-between py-5 px-4">
+            <h2 class="flex items-center text-xl font-semibold text-gray-800 dark:text-gray-200 mb-0">Workflow Records</h2>
+            <button id="pollButton" onclick="togglePolling()" class="bg-blue-500 hover:bg-blue-700 text-sm text-white py-2 px-4 rounded">
+                Enable Polling
+            </button>
+        </div>
+
+        <div class="overflow-x-auto">
+            <table class="min-w-full bg-white dark:bg-gray-900 rounded-lg shadow-sm">
+                <thead class="bg-gray-100 dark:bg-gray-700 text-gray-600 dark:text-gray-300 border-b border-gray-200 dark:border-gray-600">
+                <tr>
+                    <th class="py-3 px-4 text-center font-medium text-sm">ID</th>
+                    <th class="py-3 px-4 text-center font-medium text-sm">Foreign ID</th>
+                    <th class="py-3 px-4 text-center font-medium text-sm">Run ID</th>
+                    <th class="py-3 px-4 text-center font-medium text-sm">Run State</th>
+                    <th class="py-3 px-4 text-center font-medium text-sm">Status</th>
+                    <th class="py-3 px-4 text-center font-medium text-sm">Created At</th>
+                    <th class="py-3 px-4 text-center font-medium text-sm">Updated At</th>
+                    <th class="py-3 px-4 text-center font-medium text-sm">Object</th>
+                    <th class="py-3 px-4 text-center font-medium text-sm">Actions</th>
+                </tr>
+                </thead>
+                <tbody id="tableBody" class="divide-y divide-gray-100 dark:divide-gray-700">
+                <!-- Dynamic rows will be populated here by JavaScript -->
+                </tbody>
+            </table>
+        </div>
+    </div>
+    <div id="jsonModal" class="fixed inset-0 bg-gray-800 bg-opacity-50 flex items-center justify-center hidden">
+        <div class="bg-white p-6 rounded-lg shadow-lg min-h-[50vh] min-w-[50vw] overflow-auto">
+            <h2 class="text-xl font-semibold mb-4">Object</h2>
+            <div id="jsonContainer" class="bg-gray-100 p-4 rounded text-sm"></div>
+            <button onclick="closeModal()" class="bg-red-500 text-white px-4 py-2 mt-4 rounded">Close</button>
+        </div>
+    </div>
+</div>
+
+<script>
```

Comment:
can you pull this JS out into a separate file? i think it'll make it easier to read

Reply:
I tried this originally but then had to add a seperate handler. What I could do is put it as a separate file, embed the file, then add it as a parameter 🤔